docs/gitbook/usage/metrics.md

```diff
@@ -314,3 +314,56 @@ Reference the template in the canary analysis:
 ```
 
 **Note** that Flagger need AWS IAM permission to perform `cloudwatch:GetMetricData` to use this provider.
+
+### New Relic
+
+You can create custom metric checks using the New Relic provider.
+
+Create a secret with your New Relic Insights credentials:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: newrelic
+  namespace: istio-system
+data:
+  newrelic_account_id: your-account-id
+  newrelic_query_key: your-insights-query-key
+```
+
+New Relic template example:
+
+```yaml
+apiVersion: flagger.app/v1beta1
+kind: MetricTemplate
+metadata:
+  name: newrelic-error-rate
+  namespace: ingress-nginx
+spec:
+  provider:
+    type: newrelic
+    secretRef:
+      name: newrelic
+  query: |
+    SELECT 
+        filter(sum(nginx_ingress_controller_requests), WHERE status >= '500') / 
+        sum(nginx_ingress_controller_requests) * 100
+    FROM Metric 
+    WHERE metricName = 'nginx_ingress_controller_requests' 
+    AND ingress = '{{ ingress }}' AND  namespace = '{{ namespace }}'
```

Comment:
@stefanprodan I used the built-in prometheus query to add the `ingress = '{{ ingress }}` parameter here. However, isn't this measuring the error rate on the current primary and not on the canary?

Reply:
See https://github.com/weaveworks/flagger/issues/382#issuecomment-558573015

Reply:
One thing I noticed is that if the primary is faulty and has a high error rate, it is hard to deploy a new working version. Since the metrics are aggregated, the canary weight is not dominant enough to sufficiently reduce the error rate. Is there a good way to work around this issue when using Nginx?

Reply:
Ah nvm, just noticed that you recommend using a custom metric to instrument the application.